### DISCUSSION TITLE
Airflow 2.2.4: Not deserialising Connections

### DISCUSSION THREAD

**Discussion by kmaddi-quotient** [apache/airflow#22342] [General]
### Apache Airflow version

2.2.4 (latest released)

### What happened

I used to have fernet_key empty in 1.10.10. In the process of updating to 2.2.4 , i generated a fernet key using the given procedure in python and updated the same in aiflow.cfg. After creating a user and clicking on connections it gives me the following error.
`File "/home/airflow_user/.local/lib/python3.7/site-packages/cryptography/hazmat/backends/openssl/__init__.py", line 7, in <module>
    from cryptography.hazmat.backends.openssl.backend import backend
  File "/home/airflow_user/.local/lib/python3.7/site-packages/cryptography/hazmat/backends/openssl/backend.py", line 62, in <module>
    from cryptography.hazmat.backends.openssl.rsa import (
  File "/home/airflow_user/.local/lib/python3.7/site-packages/cryptography/hazmat/backends/openssl/rsa.py", line 422, in <module>
    class _RSAPublicKey(object):
  File "/home/airflow_user/.local/lib/python3.7/site-packages/cryptography/utils.py", line 45, in register_decorator
    def register_interface(iface):
  File "/home/airflow_user/.local/lib/python3.7/site-packages/cryptography/utils.py", line 85, in verify_interface
cryptography.utils.InterfaceNotImplemented: <class 'cryptography.hazmat.backends.openssl.rsa._RSAPublicKey'> is missing a 'recover_data_from_signature' method`

### Operating System

linux

### Versions of Apache Airflow Providers

apache-airflow-providers-apache-livy==2.2.0
apache-airflow-providers-ftp==2.0.1
apache-airflow-providers-http==2.0.3
apache-airflow-providers-imap==2.2.0
apache-airflow-providers-sqlite==2.1.0
apache-airflow-providers-ssh==2.4.1

### Deployment

Other

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template!

**Comment by potiuk** [comment]
I believe you have not used the right installatin method for 2.2.4 which uses the constraints: https://airflow.apache.org/docs/apache-airflow/stable/installation/installing-from-pypi.html

**Reply by kmaddi-quotient** [reply]
I have taken reference from https://airflow.apache.org/docs/apache-airflow/stable/upgrading-from-1-10/index.html and upgraded to 1.10.15 first. 
1. Modified necessary dags that were affecting 
2. uninstalled backports
3. upgraded to 2.2.4
In this process I had to manually make changes to airflow.cfg to accomadate new changes like moving keys from [core] to [logging] etc.

**Reply by potiuk** [reply]
Did you use constraints?

**Reply by kmaddi-quotient** [reply]
yes I did. I have used same steps as mentioned here:
`AIRFLOW_VERSION=2.2.4
PYTHON_VERSION="$(python --version | cut -d " " -f 2 | cut -d "." -f 1-2)"
CONSTRAINT_URL="https://raw.githubusercontent.com/apache/airflow/constraints-${AIRFLOW_VERSION}/constraints-${PYTHON_VERSION}.txt"
pip install "apache-airflow[async,postgres,google]==${AIRFLOW_VERSION}" --constraint "${CONSTRAINT_URL}"`

with python version as 3.7

**Reply by potiuk** [reply]
Then probably this is a problem with your base operating system and dependencies you have installed. It looks like you have a problem with "hazmat" cryptography layer not supporting all necessary cryptography functionality. Looks like you should change your openssl cryptography backend (But I am not sure how you can do it).

**Reply by potiuk** [reply]
Try different os/cryptography backend simply.
